File: 169.majority-element.js
/*
 * @lc app=leetcode id=169 lang=javascript
 *
 * [169] Majority Element
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  let res = 0;
  let cnt = 0;
  for (const num of nums) {
    if (num === res) {
      ++cnt;
    } else if (cnt === 0) {
      res = num;
      cnt = 1;
    } else {
      cnt -= 1;
    }
  }
  return res;
};
// @lc code=end

File: 169.majority-element.go
/*
 * @lc app=leetcode id=169 lang=golang
 *
 * [169] Majority Element
 */

package leetcode

// @lc code=start
func majorityElement(nums []int) int {
	res, cnt := 0, 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == res {
			cnt++
		} else if cnt == 0 {
			res, cnt = nums[i], 1
		} else {
			cnt--
		}
	}
	return res
}

// @lc code=end
